### PR TITLE
35 create glm reader

### LINKED ISSUE #35: Include Support for GOES GLM
# Requested Update

### Description
Current GeoIPS does not support the GOES Geostationary Lightning Mapper. Other operational requirements on certain systems require the inclusion of the GLM!

### Background and Motivation
Other commercial product generation and viewing programs already support GLM!

[ ] Add product for GLM
[ ] Add to GOES R series spacecraft in documentation

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [ ] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [ ] Required ***unit tests*** added and pass for new/modified functionality
* [ ] Required ***integration tests*** added and pass for new/modified functionality
* [ ] Required ***documentation*** added for new/modified functionality
* [ ] Required ***release notes*** added for new/modified functionality
* [ ] NO REQUIRED ***updates to other repos*** (explain why not required)

# Related Issues
fixes #35 

# Testing Instructions
You can run tests/integration_tests/full_test.sh to ensure that everything is still working correctly. To see output for the 'test_glm.sh' script, simply run ./tests/scripts/test_glm.sh and click the output image to see what has been produced.

# Summary
Kyle Hilburn works with GLM data, and has requested that we build a reader for that data
so he can implement his routines in GeoIPS. This branch has not only created that
reader, but also includes products, a feature_annotator, output_formatter, and a basic
algorithm to visualize GLM area products. We also modified mpl_utils.py to include a
patch adder, which adds patches to the appropriate figure in the correct Coordinate
Reference System (CRS), with the appropriate colormap applied.

    modified: geoips/geoips/image_utils/mpl_utils.py
    modified: geoips/geoips/plugins/modules/procflows/single_source.py
    modified: geoips/pyproject.toml
    added: geoips/geoips/plugins/modules/algorithms/glm_alg.py
    added: geoips/geoips/plugins/modules/output_formatters/shape_patches.py
    added: geoips/geoips/plugins/modules/readers/glm_netcdf.py
    added: geoips/geoips/plugins/yaml/products/glm.yaml
    added: geoips/tests/scripts/test_glm.sh

# Output
20230101 000000 GOES-18 glm flash_area goes_east 100p00 gov nesdis noaa 10p0
20230101 000000 GOES-18 glm group_area goes_east 100p00 gov nesdis noaa 10p0

### CODE REVIEW

File: geoips/image_utils/mpl_utils.py [hunk 1]
```diff
@@ -772,3 +774,123 @@ def create_colorbar(fig, mpl_colors_info):
...
+            crs,
+            cmap,
+            norm,
+            mapobj,
+        )
+        main_ax.add_patch(shape)
+    return main_ax
+
+
+def add_patch(lat, lon, product_val, patch, crs, cmap, norm, mapobj):
```
Comment (evrose54) [inline geoips/image_utils/mpl_utils.py:845]: We should talk about this function more to see if we want to add additional functionality for different shapes of patches. Could be useful for wind barbs, atmospheric rivers, etc.

Comment (evrose54) [reply geoips/image_utils/mpl_utils.py:845]: Issue #410 identifies the need for discussion on this point.

Comment (evrose54) [reply geoips/image_utils/mpl_utils.py:845]: Closing this comment as the issue above addresses it.

**Comment by mindyls** [comment]
@evrose54 @jsolbrig @cpcamacho This functionality has been moved to internal CIRA gitlab repo, closing this PR and @evrose54 will open another PR with the required updates to image_utils/mpl_utils.py required to make the GLM products work.

**Pull request closed without merging**